addon/collapse-key.js

```diff
@@ -5,11 +5,15 @@ export default function(property) {
     return [property];
   }
 
-  let atEachIndex = property.indexOf('.@each');
-  if (atEachIndex !== -1) {
-    return [property.slice(0, atEachIndex)];
-  } else if (property.slice(-2) === '[]') {
-    return [property.slice(0, -3)];
+  let atEachIndex = property.indexOf('@each.');
+  if (atEachIndex === -1) {
+    atEachIndex = property.indexOf('[]');
+  }
+
+  if (atEachIndex === 0) {
+    return [''];
```

Comment:
So I messed with it a bit, and I was wrong. Your empty string is the best way to handle this. Thanks for schooling me!